tests/PHPStan/Analyser/data/bug-5452.php

```diff
@@ -0,0 +1,25 @@
+<?php declare(strict_types = 1);
+
+namespace Bug5452;
+
+use function PHPStan\Testing\assertType;
+
+class Collector {
+	/** @var array<string, object> */
+	private $collection = [];
+	
+	public function rewind(): void { 
+		if (reset($this->collection) === false) {
+			assertType('array<string, object>', $this->collection);
+		}
+		if (reset($this->collection) !== false) {
+			assertType('array<string, object>&nonEmpty', $this->collection);
+		}
+		if (reset($this->collection) === null) {
```

Comment:
this null-case is tricky, because there is a generic early-return for identica-comparison against null here: https://github.com/phpstan/phpstan-src/blob/1b5710a3aa87a3d475d565c8ded12de170ee6cb6/src/Analyser/TypeSpecifier.php#L202

which does not recurse into `specifyTypesInCondition` like the true/false cases do in 

https://github.com/phpstan/phpstan-src/blob/1b5710a3aa87a3d475d565c8ded12de170ee6cb6/src/Analyser/TypeSpecifier.php#L183-L199

Reply:
just limited the PR to the actual issue reported. for now I don't know how to achieve this additional cases I imagined